zerver/openapi/zulip.yaml

```
@@ -12764,15 +12764,28 @@ paths:
                     Must be unique in the client's table of accounts.
                   type: integer
                   example: 2408
-                push_public_key:
-                  description: |
-                    Public part of the asymmetric key pair generated by the client
-                    using NaCl (the Networking and Cryptography Library), encoded
-                    using a RFC 4648 standard base64 encoder.
-
-                    It is used to encrypt notifications for delivery to the client.
+                push_key:
+                  description: |
+                    Key that the client would like the server to use to encrypt notifications. It
+                    consists of a single byte encoding which cryptosystem to use, followed by the
+                    byte sequence representing the key to use for that cryptosystem. This byte
+                    sequence is encoded using standard Base64 encoding as defined in RFC
+                    4648. Supported cryptosystems are:
```

Let's also specify the essential security properties of how the client should handle the key:
```suggestion
                    Key that the client would like the server to use to encrypt notifications,
                    encoded with Base64.

                    The key is a byte sequence beginning with a single byte that encodes which
                    cryptosystem to use, followed by the key to use for that cryptosystem. This byte
                    sequence is encoded using standard Base64 encoding as defined in RFC
                    4648.
                    
                    The client should avoid sharing the key anywhere else: in particular it should
                    generate a fresh key for each server, and to the extent possible
                    keep the key out of any backups of the client's data.
                    
                    Supported cryptosystems are:
```